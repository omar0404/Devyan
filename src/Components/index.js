import Section from './Section';
import PosterPlaceHolder from './PosterPlaceHolder';
import MovieCard from './MovieCard';
export {MovieCard, Section, PosterPlaceHolder};
