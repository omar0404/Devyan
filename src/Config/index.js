export const POSTER_BASE_URL = 'http://image.tmdb.org/t/p/w500';
