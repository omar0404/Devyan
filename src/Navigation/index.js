import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import HomeScreens from './HomeScreens';

const RootNavigation = () => (
  <NavigationContainer>
    <HomeScreens />
  </NavigationContainer>
);
export default RootNavigation;
