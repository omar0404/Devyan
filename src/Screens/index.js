import Home from './Home';
import AddMovie from './AddMovie';

export {Home, AddMovie};
