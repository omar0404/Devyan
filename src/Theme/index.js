import Colors from './Colors';

const theme = {
  backgroundColor: Colors.black,
  text: {
    primary: Colors.white,
    secondary: Colors.gray,
    teritary: Colors.lightPink,
  },
  button: {
    default: Colors.lightPink,
    disabled: Colors.gray,
  },
  loadingIndicator: Colors.lightPink,
  posterPlaceHolderBorder: Colors.gray,
  textInputBorder: Colors.gray,
  textInputPlaceHolder: Colors.gray,
  headerShadowColor: Colors.shark,
  sectionLine: Colors.lightPink,
  addMovieIcon: Colors.lightPink,
  headerTintColor: Colors.gray,
};
export default theme;
