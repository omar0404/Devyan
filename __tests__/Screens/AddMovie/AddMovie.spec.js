import React from 'react';
import {create} from 'react-test-renderer';
import {AddMovie} from 'MoviesApp/src/Screens';
import {
  handleInputChange,
  handleMoviePosterUpload,
  handleAddMoviePress,
} from 'MoviesApp/src/Screens/AddMovie/AddMovie';

describe('AddMovie Screen', () => {
  it('should render correctly', () => {
    const wrapper = create(<AddMovie />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should update form values state when input changes', () => {
    const formValues = {};
    const setFormValues = jest.fn((cb) => {
      cb(formValues);
    });
    const fieldName = 'title';
    const fieldValue = 'Batman';
    handleInputChange(fieldName, setFormValues)(fieldValue);
    expect(setFormValues).toBeCalled();
  });
  it('should update form values state when poster movie uploaded', () => {
    const formValues = {};
    const imageUri = 'fakeUri';
    const setFormValues = jest.fn((cb) => {
      cb(formValues);
    });
    handleMoviePosterUpload(setFormValues)(imageUri);
    expect(setFormValues).toBeCalled();
  });
  it('should navigate to home screen with added movie', () => {
    const formValues = {};
    const expectedParams = {addedMovie: formValues};
    const navigation = {navigate: jest.fn()};
    handleAddMoviePress(navigation, formValues)();
    expect(navigation.navigate).toHaveBeenCalledWith('Home', expectedParams);
  });
});
