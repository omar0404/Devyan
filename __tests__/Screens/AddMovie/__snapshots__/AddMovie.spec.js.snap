// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie Screen should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "gray",
          "borderStyle": "dotted",
          "borderWidth": 1,
          "flex": 1,
          "flexBasis": 40,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "gray",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Entypo",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï‡‡
      </Text>
      <Text
        style={
          Object {
            "color": "gray",
            "marginTop": 12,
            "textAlign": "center",
          }
        }
      >
         Upload Movie
        

        Poster
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Title *"
        placeholderTextColor="gray"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "borderRadius": 5,
            "color": "gray",
            "fontSize": 18,
            "height": 45,
            "marginTop": 40,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Overview"
        placeholderTextColor="gray"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "borderRadius": 5,
            "color": "gray",
            "fontSize": 18,
            "height": 45,
            "marginTop": 40,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Date"
        placeholderTextColor="gray"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "borderRadius": 5,
            "color": "gray",
            "fontSize": 18,
            "height": 45,
            "marginTop": 40,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "gray",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 50,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Add Movie
    </Text>
  </View>
</View>
`;
